packages/ruleset/src/functions/resource-response-consistency.js

```diff
@@ -0,0 +1,217 @@
+ * return the same schema as the "GET" request for the resource.
+ *
+ * An operation is considered to be a "create" operation if the operationId
+ * starts with "create" OR it's a POST request and there is a similar path
+ * but with a trailing path parameter reference.
+ *
+ * An operation is considered to be an "update" operation if the method type
+ * is "put" or "patch".
+ *
+ * @param {*} operation an operation within the API definition
+ * @param {*} path the array of path segments indicating the "location" of the operation within the API definition
+ * @param {*} apidef the resolved API spec
+ * @returns an array containing the violations found or [] if no violations
+ */
+function resourceResponseConsistency(operation, path, apidef) {
+  logger.debug(
+    `${ruleId}: checking responses for operation at location: ${path.join('.')}`
+  );
+
+  // This rule only applies to POSTs if they are determined to be "create" operations.
+  if (
+    isOperationOfType('post', path) &&
+    !isCreateOperation(operation, path, apidef)
+  ) {
+    // In this case, no need to check the responses.
+    logger.debug(
+      `${ruleId}: operation is not a "create", PUT, or PATCH - no need to check`
+    );
+    return [];
+  }
+
+  const canonicalSchema = getCanonicalSchema(path, apidef);
+
+  // If the GET request does not define a schema in its response, there is nothing left to check.
+  if (!canonicalSchema) {
+    logger.debug(
+      `${ruleId}: no response schema found for GET request - abandoning check for this operation`
+    );
+    return [];
+  }
+
+  // If there are no responses defined, there is nothing to check.
+  // We validate for that issue elsewhere.
+  if (!operation.responses) {
+    logger.debug(
+      `${ruleId}: no responses object found on operation at location: ${path.join(
+        '.'
+      )}`
+    );
+    return [];
+  }
+
+  const [, successCodes] = getResponseCodes(operation.responses);
+  logger.debug(
+    `${ruleId}: operation has the following successCodes codes: ${successCodes.join(
+      ', '
+    )}`
+  );
+
+  // If there are no success code responses, there is nothing to check.
+  // We validate for that issue elsewhere.
+  if (!successCodes || !successCodes.length) {
+    logger.debug(
+      `${ruleId}: no success codes found on operation at location: ${path.join(
+        '.'
+      )}`
+    );
+    return [];
+  }
+
+  const errors = [];
+
+  // Check defined, synchronous success responses for schema content that is equivalent
+  // to the canonical schema, as defined by the corresponding GET request.
+  // Appropriate use of other codes that don't define body content, like 202 or 204,
+  // is validated for elsewhere.
+  ['200', '201'].forEach(code => {
+    if (successCodes.includes(code)) {
+      const successResponse = operation.responses[code];
+      if (
+        isObject(successResponse) &&
+        successResponse.content &&
+        successResponse.content['application/json'] &&
```

Comment:
> I think this logic breaks for a mimetype like application/json; charset=utf-8...

Great catch, I will make that change

> Another thing to think about... do we perhaps want to expand our to include non-JSON responses as well? Not sure what the implications of that might be.

I don't think so. Since we're explicitly checking schemas (instances of JSON Schema), I think this only applies to JSON response content

Reply:
This has now been addressed